### PR TITLE
Add smpp-connector-example.md

### PR DESCRIPTION
**Pull request by KasunAratthanage**

### CODE REVIEW

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+The following diagram shows the overall solution. There is HTTP API and user can invoke it with http call with JSON. API is able to send a SMS for the request number in JSON request with message in JSON. 
+
+<img src="../../../../assets/img/connectors/smpp-connector-example.png" title="SMPP connector example" width="800" alt="smpp connector example"/>
+
+If you do not want to configure this yourself, you can simply [get the project](#get-the-project) and run it.
+
+## Configure the connector in WSO2 Integration Studio
+
+Follow these steps to set up the ESB Solution Project and the Connector Exporter Project.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:24]:
```suggestion
Follow these steps to set up the Integration Project and the Connector Exporter Project.
```

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+Given below is a sample scenario that demonstrates how to work WSO2 EI SMPP Connector and send SMS messages via the SMPP protocol.
+
+The SMPP server in SMSC have all ESME (External Short Messaging Entity) addresses which is an external application that connects to a SMSC and the active connection. When you send SMS to destination, it comes to the SMSC, then one of the module in SMSC checks the destination address whether it is available or not. If it is available it create connection object that is responsible for sending SMS message.
+There are many SMPP gateways available in the world.and now almost all the Message centers support SMPP. It is not practical always to connect with real SMSC. However in this scenario we will try it with [SMSC simulator](https://github.com/smn/logica-smpp-sim).
+
+The following `sendSMS`operation is exposed via an API. The API with the context `/send` has one resource.
+
+* `/send` : Use to send SMS Message to the Short Message service center.
+
+The following diagram shows the overall solution. There is HTTP API and user can invoke it with http call with JSON. API is able to send a SMS for the request number in JSON request with message in JSON.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:16]:
```suggestion
The following diagram shows the overall solution. There is an HTTP API that you can invoke with an HTTP call with JSON. The API is able to send a SMS for the request number in a JSON request with the message in JSON. 
```

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
+# SMPP Connector Example
+
+SMPP (Short Message Peer to Peer protocol) Connector allows you to send SMS through the WSO2 EI. It uses [jsmpp API](https://jsmpp.org/) to communicate with a SMSC (Short Message service center) which is to store, forward, convert and deliver Short Message Service (SMS) messages. jsmpp is a java implementation of SMPP protocol. 
+
+## What you'll build
+
+Given below is a sample scenario that demonstrates how to work WSO2 EI SMPP Connector and send SMS messages via the SMPP protocol.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:7]:
```suggestion
Given below is a sample scenario that demonstrates how to work with the WSO2 EI SMPP Connector and send SMS messages via the SMPP protocol.
```

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+The SMPP server in SMSC have all ESME (External Short Messaging Entity) addresses which is an external application that connects to a SMSC and the active connection. When you send SMS to destination, it comes to the SMSC, then one of the module in SMSC checks the destination address whether it is available or not. If it is available it create connection object that is responsible for sending SMS message.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:9]:
```suggestion
The SMPP server in SMSC have all the ESME (External Short Messaging Entity) addresses. This is an external application that connects to a SMSC and the active connection. When you send an SMS to a destination, it comes to the SMSC. Then one of the modules in SMSC checks if the destination address is available or not. If it is available, it creates a connection object that is responsible for sending the SMS message.
```

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+There are many SMPP gateways available in the world.and now almost all the Message centers support SMPP. It is not practical always to connect with real SMSC. However in this scenario we will try it with [SMSC simulator](https://github.com/smn/logica-smpp-sim).
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:10]:
```suggestion
There are many SMPP gateways available in the world and now almost all the message centers support SMPP. It is not practical always to connect with real SMSC. However, in this scenario we will try it with [SMSC simulator](https://github.com/smn/logica-smpp-sim).
```

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+The following `sendSMS`operation is exposed via an API. The API with the context `/send` has one resource.
+
+* `/send` : Use to send SMS Message to the Short Message service center.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:14]:
```suggestion
* `/send` : Used to send SMS Message to the Short Message Service Center.
```

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
+# SMPP Connector Example
+
+SMPP (Short Message Peer to Peer protocol) Connector allows you to send SMS through the WSO2 EI. It uses [jsmpp API](https://jsmpp.org/) to communicate with a SMSC (Short Message service center) which is to store, forward, convert and deliver Short Message Service (SMS) messages. jsmpp is a java implementation of SMPP protocol.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:3]:
```suggestion
SMPP (Short Message Peer-to-Peer Protocol) Connector allows you to send an SMS through the WSO2 EI. It uses the [jsmpp API](https://jsmpp.org/) to communicate with an SMSC (Short Message Service Center), which is used to store, forward, convert, and deliver Short Message Service (SMS) messages. jsmpp is a Java implementation of the SMPP protocol. 
```

Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:3]: Updated.

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md (hunk 3 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:7]: Updated.

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md (hunk 4 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:9]: Updated.

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md (hunk 5 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:10]: Updated as following manner.
There are many SMPP gateways available in the world and now almost all the message centers support SMPP. It is not practical always to connect with real SMSC. However, in this scenario we will try it with **SMSC simulator**. Please refer the [Setting up SMPP connector](setting-up-smpp-connector.md) documentation.

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md (hunk 6 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:14]: Updated.

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md (hunk 2 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:16]: Updated.

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md (hunk 1 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:24]: Updated.

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+
+Create a resource to send an SMS to the Short Message Service Center.
+
+1. Initialize the connector.
+    
+    1. Navigate into the **Palette** pane and select the graphical operations icons listed under **SMPP Connector** section. Then drag and drop the `init` operation into the Design pane.
+        
+        <img src="../../../../assets/img/connectors/smpp-drag-and-drop-init.png" title="Drag and drop init operation" width="500" alt="Drag and drop init operation"/>   
+    
+    2. Add the property values into the `init` operation as shown bellow. Replace the `host`, `port`, `systemId`, `password` with your values.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:48]:
```suggestion
    2. Add the property values into the `init` operation as shown below. Replace the `host`, `port`, `systemId`, `password` with your values.
```

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+               
+        - **sourceAddress** : Source address of the SMS message. 
+        - **destinationAddress** : Destination address of the SMS message. 
+        - **message** : Content of the SMS message.
+        
+        While invoking the API, the above three parameters values come as a user input.
+        
+        <img src="../../../../assets/img/connectors/smpp-drag-and-drop-sendSMS-parameters.png" title="Drag and drop send operation" width="500" alt="Drag and drop send operation"/> 
+    
+    3. To get the input values in to the API we can use the [property mediator](https://ei.docs.wso2.com/en/next/micro-integrator/references/mediators/property-Mediator/). Navigate into the **Palette** pane and select the graphical mediators icons listed under **Mediators** section. Then drag and drop the `Property` mediators into the Design pane as shown bellow.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:76]: Please use a relative link here.

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+    6. Add the property mediator to capture the `distinationAddress` values. The message contains content of the SMS message.                  
+       
+        - **name** : distinationAddress
+        - **expression** : json-eval($.distinationAddress)
+         
+        <img src="../../../../assets/img/connectors/smpp-api-property-mediator-property3-value3.png" title="Add values to capture distinationAddress" width="600" alt="Add values to capture distinationAddress"/>  
+        
+3. Get a response from the user.
+    
+    When you invoking the created API the request of the message is going through the `/send` resource. Finally pass to the the [Respond mediator](https://ei.docs.wso2.com/en/next/micro-integrator/references/mediators/respond-Mediator/). In here the Respond Mediator stops the processing on the current message and sends the message back to the client as a response.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:107]:
```suggestion
    When you are invoking the created API, the request of the message is going through the `/send` resource. Finally, it is passed to the [Respond mediator](https://ei.docs.wso2.com/en/next/micro-integrator/references/mediators/respond-Mediator/). The Respond Mediator stops the processing on the current message and sends the message back to the client as a response.            
```

Add a relative link here.

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+    
+    1. Drag and drop **respond mediator** to the **Design view**. 
+    
+         <img src="../../../../assets/img/connectors/smpp-drag-and-drop-respond-mediator.png" title="Add Respond mediator" width="800" alt="Add Respond mediator"/> 
+
+    2. Once you have setup the sequences and API, you can see the `salesforcerest` API as shown below.
+    
+         <img src="../../../../assets/img/connectors/smpp-api-design-view.png" title="API Design view" width="600" alt="API Design view"/>
+        
+    > **Note**: That the properties should be add to the pallet before create the operation.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:117]:
```suggestion
    > **Note**: The properties should be added to the pallet before creating the operation.
```

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md (hunk 8 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:48]: Updated.

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md (hunk 10 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:107]: Updated.

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md (hunk 11 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:117]: Updated.

File: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md (hunk 9 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-example.md:76]: Updated.

**Pull request merged**